client/my-sites/people/team-list/test/team.jsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import useFakeDom from 'test/helpers/use-fake-dom';
+import useMockery from 'test/helpers/use-mockery';
+
+describe( 'Team', function() {
+	let Card;
+	let PeopleListSectionHeader;
+	let InfiniteList;
+	let Team;
+	let configMock;
+	let deterministicStringify;
+
+	useFakeDom();
+	useMockery( mockery => {
+		configMock = sinon.stub();
+		configMock.isEnabled = sinon.stub();
+		mockery.registerMock( 'config', configMock );
+
+		deterministicStringify = sinon.stub();
+		mockery.registerMock( 'lib/deterministic-stringify', deterministicStringify );
+		mockery.registerMock( 'lib/sites-list', () => null );
+	} );
+
+	before( function() {
+		Card = require( 'components/card' );
+		PeopleListSectionHeader = require( 'my-sites/people/people-list-section-header' ).default;
+		InfiniteList = require( 'components/infinite-list' );
+
+		Team = require( '../team' ).Team;
+	} );
+
+	it( 'renders infinite list when site is specified and any users have been loaded', function() {
+		deterministicStringify
+			.withArgs( { option1: 'option1value', option2: 'option2value' } )
+			.returns( 'list key' );
+
+		const wrapper = shallow(
+			<Team
+				fetchInitialized={ true }
+				fetchingUsers={ false }
+				fetchOptions={ { option1: 'option1value', option2: 'option2value', number: 10, offset: 2 } }
+				site={ 'site1' }
+				users={ [ 'user1' ] }
+				excludedUsers={ [] }
+			/>
+		);
+
+		expect( wrapper.find( Card ).find( InfiniteList ).key() ).to.equal( 'list key' );
+	} );
+
+	it( 'renders translated team header text by default', function() {
+		const translate = sinon.stub()
+			.withArgs( 'Team' )
+			.returns( 'team label' );
+
+		const wrapper = shallow(
+			<Team
+				fetchInitialized={ true }
+				fetchingUsers={ false }
+				fetchOptions={ {} }
+				users={ [] }
+				excludedUsers={ [] }
+				translate={ translate }
+			/>
+		);
+
+		expect( wrapper.find( PeopleListSectionHeader ).prop( 'label' ) )
+			.to.equal( 'team label' );
+	} );
+
+	it( 'renders translated header text for search when there are search results', function() {
+		const translate = sinon.stub()
+			.returns( 'translated search label' );
+
+		const wrapper = shallow(
+			<Team
+				fetchInitialized={ true }
+				fetchingUsers={ false }
+				fetchOptions={ {} }
+				search={ 'abc' }
+				site={ 'site1' }
+				totalUsers={ 1 }
+				users={ [ 'user1' ] }
+				excludedUsers={ [] }
+				translate={ translate }
+			/>
+		);
+
+		expect( wrapper.find( PeopleListSectionHeader ).prop( 'label' ) )
+			.to.equal( 'translated search label' );
+	} );
```

Comment:
We will no doubt have many entertaining conversations on this topic :)

My opinions on unit testing over the years have only become more vague and adaptable to circumstances.  I want certain characteristics: confidence that I haven't broken something when I make changes and also I want to be able to make changes without feeling unnecessarily obstructed by existing tests.  I also want to learn the set of testing conventions the team is converging on (just so there are fewer decisions to make when writing a new component).

As far as duplicating content between tests and implementation - i'm happy to go with the majority.

The goal of this PR is to add more than 0 test coverage to a couple of components.  Once all feedback has been addressed, this PR may be used as a guide for updating the many other components that have no test coverage (and do not use ES6, redux for analytics, translation, etc.).